refactor: add root command to get the root of the current repository

AdeAttwood opened the pull request:
refactor: add root command to get the root of the current repository

Summary:

Right now in our git hooks we are using `git rev-parse --show-toplevel` to get
the root of the repository. This is not working within a sapling repo.

This implements the root command so we can use that instead. It as a new `root`
method to the `SourceControlProvider` interface. This will then be implemented
by any provider we want to.

Test Plan:

No real way to test this other than running the command and checking the
output. As long as the hooks commands will be working for us this will be fine
for now.

---
[//]: # (BEGIN SAPLING FOOTER)
Stack created with [Sapling](https://sapling-scm.com). Best reviewed with [ReviewStack](https://reviewstack.dev/Practically/Conventional-Tools/pull/90).
* __->__ #90
* #89
* #86
* #85

GeorgeCadwallader commented on src/commands/root.ts:
```diff
@@ -0,0 +1,17 @@
+    throw new Error('No source control provider found');
+  }
+
+  console.log(await sourceControl.root());
```
Do we not want to use our "log" function for this?

I know we mainly created it for spying purposes in the testing but we might want to test this in the future and it promotes consistency?

AdeAttwood committed:
refactor: add root command to get the root of the current repository

Summary:

Right now in our git hooks we are using `git rev-parse --show-toplevel` to get
the root of the repository. This is not working within a sapling repo.

This implements the root command so we can use that instead. It as a new `root`
method to the `SourceControlProvider` interface. This will then be implemented
by any provider we want to.

Test Plan:

No real way to test this other than running the command and checking the
output. As long as the hooks commands will be working for us this will be fine
for now.
